### PR TITLE
feat(editor): Add template Id to workflow metadata

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
Adding a link between the workflow and the template it originated from by saving `templateId` in the workflow metadata

## Related tickets and issues
ADO-1537

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: cypress/e2e/29-templates.cy.ts [hunk 1]
```diff
@@ -29,12 +29,33 @@ describe('Templates', () => {
...
+
+		cy.url().then(($url) => {
+			const templateId = $url.split('/').pop();
+
+			templatesPage.getters.useTemplateButton().click();
+			cy.url().should('include', '/workflow/new');
+			workflowPage.actions.saveWorkflowOnButtonClick();
+
+			cy.wait('@saveWorkflow').then((interception) => {
+				expect(interception.request.body.meta.templateId).to.equal(templateId);
```
Comment (mutdmour) [inline cypress/e2e/29-templates.cy.ts:52]: is this really a good e2e test? seems to be more of an integration test. it's too coupled to the api request and does not tell you it was stored or used correctly.. Could you think of a better approach, even though it does not seem we are using this yet.. maybe check the json of loaded workflow or when copying/pasting?

File: packages/editor-ui/src/Interface.ts [hunk 2]
```diff
@@ -272,6 +271,8 @@ export interface INewWorkflowData {
 
 export interface WorkflowMetadata {
 	onboardingId?: string;
+	templateId?: string;
+	instanceId?: string;
```
Comment (mutdmour) [inline packages/editor-ui/src/Interface.ts:275]: are we adding instance id now to the metadata? we intentionally avoided adding that before when saving.. only to be added when sharing..

Comment (MiloradFilipovic) [reply packages/editor-ui/src/Interface.ts:275]: We are not adding `instanceId` on any new places in this PR, I moved it here so we can re-use this type in `IWorkflowToShare` and clean up the typing around that a bit more.

File: cypress/e2e/29-templates.cy.ts (hunk 1 above)
Comment (MiloradFilipovic) [reply cypress/e2e/29-templates.cy.ts:52]: Good point on the API coupling. The problem with meta is that it has to rely for something under the hood (like you mentioned). Goal here is to test if template user chooses is correctly saved, but I guess we can check that also by re-opening the workflow, will try something like that.

Comment (mutdmour) [reply cypress/e2e/29-templates.cy.ts]: ya this might be one of those scenarios where a unit test is better.. but nice solution here..

**Review by mutdmour — APPROVED** [review]

**Comment by Joffcom** [comment]
Got released with [`n8n@1.23.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.23.0)

**Pull request merged**